node-backend/app/controllers/v1/recognitionController.js

```diff
@@ -0,0 +1,326 @@
+const yup = require("yup");
+const moment = require("moment"); //eslint-disable-line node/no-extraneous-require
+
+const db = require("../../models/sequelize");
+const jwtValidate = require("../../jwtTokenValidation/jwtValidation");
+const utility = require("../../utils/utility");
+const Recognitions = db.recognitions;
+const CoreValues = db.coreValues;
+const Users = db.users;
+
+const schema = yup.object().shape({
+  core_value_id: yup
+    .number({ core_value_id: "should be number" })
+    .required({ core_value_id: "required" })
+    .typeError({ core_value_id: "should be number" }),
+  text: yup.string({ text: "should be text" }).required({ text: "required" }),
+  given_for: yup
+    .number()
+    .required({ given_for: "required" })
+    .typeError({ given_for: "should be number" }),
+  given_by: yup.number().typeError({ given_by: "should be number" }),
+  given_at: yup.number().typeError({ given_at: "should be number" }),
+});
+
+exports.create = /*eslint-disable-line node/exports-style*/ async (
+  req,
+  res
+) => {
+  const authHeader = req.headers["authorization"];
+  const tokenData = await jwtValidate.getData(authHeader);
+  // Create a Recognition
+  const recognitions = {
+    core_value_id: req.body.core_value_id,
+    text: req.body.text,
+    given_for: req.body.given_for,
+    given_by: tokenData.userId,
+    given_at: moment.utc().unix(),
+  };
+  // Validate request
+  schema
+    .validate(recognitions, { abortEarly: false })
+    .then(() => {
```

Comment:
break down this code in modular functions. Like validating corevalue should be a separate function in this file. Validating given_for user should be a different function. And then you can call these functions in this mainn function